include/jemalloc/internal/spin_delay_arm.h

```diff
@@ -0,0 +1,30 @@
+#include "jemalloc/internal/jemalloc_preamble.h"
+#include <sys/auxv.h>
+
+/* Define HWCAP_SB if not already defined in system headers */
+#ifndef HWCAP_SB
+#define HWCAP_SB (1UL << 56) /* Speculation Barrier */
+#endif
+
+
+/* Use SB instruction if available, otherwise ISB */
+static inline void
+spin_delay_arm(void) {
+	static int use_spin_delay_sb = -1;
+
+	if (__builtin_expect(use_spin_delay_sb, -1) == 1) {
+		/* SB instruction encoding */
+		asm volatile(".inst 0xd50330ff \n");
+	} else if (use_spin_delay_sb == 0) {
+		/* ISB instruction */
+		asm volatile("isb");
+	} else {
+		/* Initialize variable and check if SB is supported */
+		if (getauxval(AT_HWCAP) & HWCAP_SB) {
+			use_spin_delay_sb = 1;
+		} else {
+			use_spin_delay_sb = 0;
+		}
```

Comment:
This seems like the type of thing that could happen in a `__attribute__((constructor))` so that there never needs to be the fallback path in the function.